File: src/main/java/com/xebia/xboson/NameSelectionRule.java
package com.xebia.xboson;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.Optional;

/**
 * Created by xviuda on 22-01-16.
 */
public class NameSelectionRule implements SelectionRule {
    @Override
    public Optional<ParsedElement> match(Document doc, Element element) {
        String tagName = element.tagName();
        String name = element.attr("name");
        if (name == null) {
            return Optional.empty();
        }
        String cssQuery = String.format("%s[name=%s]", tagName, name);
        Elements found = doc.select(cssQuery);
        if (found.size() == 1) {
            return Optional.of(new ParsedElement(name, cssQuery));
        }else {
            return Optional.empty();
        }
    }
}

File: src/main/kotlin/com/xebia/xboson/NameSelectionRule.kt
package com.xebia.xboson

import org.jsoup.nodes.Document
import org.jsoup.nodes.Element

class NameSelectionRule : SelectionRule {
    override fun match(document: Document, element: Element): ParsedElement? {
        val tagName = element.tagName()
        val name = element.attr("name")
        if ("" == name) {
            return null;
        }
        val cssQuery = String.format("%s[name=%s]", tagName, name)
        val found = document.select(cssQuery)
        if (found.size == 1) {
            return ParsedElement(name, cssQuery)
        } else {
            return null
        }
    }
}
